org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -2325,9 +2343,14 @@ public boolean updateAnnotationNullAnalysisOptions(IJavaProject javaProject, boo
 		if (enabled) {
 			String nonnullType = getAnnotationType(javaProject, this.nonnullTypes, nonnullClasspathStorage);
 			String nullableType = getAnnotationType(javaProject, this.nullableTypes, nullableClasspathStorage);
-			projectNullAnalysisOptions = generateProjectNullAnalysisOptions(nonnullType, nullableType);
+			String nonnullbydefaultTypes = getAnnotationType(javaProject, this.nonnullbydefaultTypes, nonnullbydefaultClasspathStorage);
+			if (nonnullbydefaultTypes == null && nonnullType != null && nullableType != null) {
+				// there is not NonNullByDefault in org.jetbrains:annotations
+				nonnullbydefaultTypes = "org.eclipse.jdt.annotation.NonNullByDefault";
```

Comment:
Wouldn't this code auto-enable the feature even when the user may have explicitly intended to not enable it ? Maybe they just want non-null/null annotations but not to have it work by default.

Reply:
The user does not have to use `NonNullByDefault`. It's just a default property if nothing is set.
Eclipse works the same way. A fully-qualified type name is required, `org.eclipse.jdt.annotation.NonNullByDefault` by default.

Reply:
Right. Needed to read up on how to use this properly. Looks fine now.